aten/src/ATen/core/ivalue_inl.h

```diff
@@ -494,6 +494,31 @@ struct C10_EXPORT ivalue::Future final : c10::intrusive_ptr_target {
     return childFut;
   }
 
+  c10::intrusive_ptr<Future> thenAsync(
```

Comment:
Looks like there is no usage of this API in this stack. Do we plan to use it in the near future?